service/builder/exporters_builder.go

```diff
@@ -173,6 +174,12 @@ func (eb *ExportersBuilder) buildExporter(
 			return nil, fmt.Errorf("error creating %s exporter: %v", config.Name(), err)
 		}
 
+		// The factories can be implemented by third parties, check if they really
```

Comment:
Nit: I think it is not necessary to mention third parties here. It is sufficient to say that a factory may work incorrectly and return nil error and nil exporter and we want to handle that case correctly.

Reply:
Good point.